.github/workflows/release.yaml

```diff
@@ -0,0 +1,243 @@
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+
+      - name: Install Node.js ${{ env.NODEJS_VERSION }}
+        if: matrix.platform == 'js'
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ env.NODEJS_VERSION }}
+
+      - name: Cache protobuf ${{ env.PROTOBUF_VERSION }}
+        uses: actions/cache@v2
+        with:
+          path: ~/protobuf
+          key: ${{ runner.os }}-protobuf-${{ env.PROTOBUF_VERSION }}
+
+      - name: Download conformance test files
+        uses: actions/download-artifact@v2
+        with:
+          name: conformance-test-files
+
+      - name: Extract conformance test files
+        run: tar -xvf conformance-test-files.tar
+
+      - name: Run ${{ matrix.platform }} conformance tests
+        run: |
+          export CONF_TEST_PATH=~/protobuf/conformance/conformance-test-runner
+          ./conformance/test-conformance.sh ${{ matrix.platform }}
+
+  build-examples:
+    needs: build
+    runs-on: ubuntu-18.04
+
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Download maven artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: maven-artifacts
+          path: ~/.m2/repository
+
+      - name: Cache gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-examples-${{ hashFiles('examples/**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-examples-
+
+      - name: Install Java ${{ env.JAVA_VERSION }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+
+      - name: Build all examples
+        run: |
+          for d in examples/*; do
+            set -e
+            [ -d "$d" ] || continue
+            echo "::group::Build $d"
+            pushd "$d" >/dev/null
+            ../../gradlew build
+            popd >/dev/null
+            echo "::endgroup::"
+          done
+
+  publish:
+    needs: [build, conformance, build-examples]
+    runs-on: ubuntu-18.04
+
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Cache gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+            ~/.konan
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Install Java ${{ env.JAVA_VERSION }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+
+      - name: Publish to Bintray
+        run: ./gradlew clean publishToBintrayRepository
```

Comment:
I'm not sure why a `clean` would be needed here, but as this was already happening on CircleCI I decided to keep it.

Reply:
The restored gradle cache sometimes has files from previous builds that cause build breakage or cause gradle to fail to rebuild a file that did in fact change. So the `clean` is an extra safety measure to make sure the released code isn't accidentally being built with cached outputs that are incorrect for this build.